libs/state/src/lib/cell-summary/cell-summary.state.ts

```diff
@@ -19,13 +21,20 @@ export class CellSummaryState {
   /** Data service to load the FTU data */
   private readonly dataService = inject(FtuDataService);
 
+  readonly sources = selectSnapshot(ActiveFtuSelectors.sources);
+
+  readonly iri = selectSnapshot(ActiveFtuSelectors.iri);
+
+  readonly setIri = dispatch(ActiveFtuActions.SetIri);
+
   /**
    * Loads the cell summary data and aggregrated of the current Iri into
    * the state and cancels uncompleted action if any
    */
   @Action(Load, { cancelUncompleted: true })
   load({ patchState, dispatch }: Context, { iri }: Load): Observable<unknown> {
-    return this.dataService.getCellSummaries(iri).pipe(
+    this.setIri(iri);
```

Comment:
You have to wait for this dispatch to finish before accessing `this.sources()` otherwise you might introduce race conditions.
```ts
const loadData = () => this.dataService.getCellSummaries(...).pipe(...)
return this.setIri(iri).pipe(switchMap(loadData))
```